components/TARGET_PSA/services/crypto/COMPONENT_SPE/psa_crypto_partition.c

```diff
@@ -328,22 +330,27 @@ static void psa_mac_operation(void)
 
                     uint8_t *mac = mbedtls_calloc(1, mac_length);
                     if (mac == NULL) {
+                        psa_mac_abort(msg.rhandle);
```

Comment:
won't it cause recursion?
calling psa_mac_abort will call psa_mac_abort which will call this function with case of abort, which calls psa_mac_abort in case of abort

Reply:
No, this is service code calling library code, the library never calls the service.